### PR TITLE
enable lint in CI, use a more modular approach to allow reuse

### PR DESCRIPTION
**Pull request by ldruschk**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -7,6 +7,9 @@ trigger:
 - master
 - dev
 
+pr:
+- master
```
Comment (Trolldemorted) [inline azure-pipelines.yml:11]: This is active by default, isn't it?

**Comment by ldruschk** [comment]
Not sure, I assumed it is not

**Comment by Trolldemorted** [comment]
>Not sure, I assumed it is not

This PR was successfully built by azure, and I think the documentation says somewhere that PRs are enabled by default

**Comment by Savallator** [comment]
I don't get why we should bloat the lines so much? I personally think the readability is much worse with these changes... after all, people usually have monitors wider than a nokia 3310...
Otherwise i think its fine...

**Comment by ldruschk** [comment]
That's just the black default settings and explicitly set for isort to match this. We can simply adjust the line length and run black again

**Comment by Trolldemorted** [comment]
To increase the line width we have to increase `line_length=XX` in the ` .isort.cfg`, `max-line-length=XX` somewhere in the flake8 config, and `--line-length` in the black params.

Did I forget anything @ldruschk ?

**Comment by ldruschk** [comment]
You don't need to adjust flake8 since only the pyflakes stuff is enabled, black and isort should be enough

**Pull request merged**